speed-dreams/source-2.2.3/src/simulatedDrivingAssistance/backendUnitTests/DecisionsTests.cpp

```diff
@@ -0,0 +1,121 @@
+#include <gtest/gtest.h>
+#include "TestUtils.h"
+#include "ConfigEnums.h"
+#include "Mediator.h"
+#include "Mediator.inl"
+#include "mocks/DecisionMakerMock.h"
+#include "../rppUtils/Random.hpp"
+#include "../rppUtils/RppUtils.hpp"
+#include "IndicatorConfig.h"
+
+/// @brief Initialize SMediator with a car which values are set to 0
+void InitializeMediator()
+{
+    SetupSingletonsFolder();
+
+    tCarElt car;
+    car.ctrl.brakeCmd = 0;
+    car.ctrl.accelCmd = 0;
+    car.ctrl.steer = 0;
+
+    CarController carController;
+    carController.SetCar(&car);
+
+    carController.SetBrakeCmd(0);
+    carController.SetAccelCmd(0);
+    carController.SetSteerCmd(0);
+
+    SMediator::GetInstance()->CarController = carController;
+}
+
+/// @brief Tests if all decisions do their RunInterveneCommand correctly
+TEST(DecisionsTest, RunInterveneDecisions)
+{
+    InitializeMediator();
+
+    Random random;
+
+    BrakeDecision brakeDecision;
+    float controlBrakeAmount = random.NextFloat();
+    brakeDecision.BrakeAmount = controlBrakeAmount;
+    brakeDecision.RunInterveneCommands();
+
+    std::cout << "Testing brake...";
+    ASSERT_ALMOST_EQ(controlBrakeAmount, SMediator::GetInstance()->CarController.GetBrakeCmd(), 0.001f);
+    std::cout << " check" << std::endl;
+
+    AccelDecision accelDecision;
+    float controlAccelAmount = random.NextFloat();
+    accelDecision.AccelAmount = controlAccelAmount;
+    accelDecision.RunInterveneCommands();
+
+    std::cout << "Testing accel...";
+    ASSERT_ALMOST_EQ(controlAccelAmount, SMediator::GetInstance()->CarController.GetAccelCmd(), 0.001f);
+    std::cout << " check" << std::endl;
+
+    SteerDecision steerDecision;
+    float controlSteerAmount = random.NextFloat();
+    steerDecision.SteerAmount = controlSteerAmount;
+    steerDecision.RunInterveneCommands();
+
+    std::cout << "Testing steer...";
+    ASSERT_ALMOST_EQ(controlSteerAmount, SMediator::GetInstance()->CarController.GetSteerCmd(), 0.001f);
+    std::cout << " check" << std::endl;
+
+    std::cout << "Checking if no value was changed that should not have been changed...";
+    ASSERT_ALMOST_EQ(controlBrakeAmount, SMediator::GetInstance()->CarController.GetBrakeCmd(), 0.001f);
+    ASSERT_ALMOST_EQ(controlAccelAmount, SMediator::GetInstance()->CarController.GetAccelCmd(), 0.001f);
+    ASSERT_ALMOST_EQ(controlSteerAmount, SMediator::GetInstance()->CarController.GetSteerCmd(), 0.001f);
+    std::cout << " check" << std::endl;
+}
+
+/// @brief Checks if the brake decision RunIndicateCommand works correctly
+TEST(DecisionTests, BrakeRunIndicateTest)
+{
+    InitializeMediator();
+    GfInit();
+
+    BrakeDecision brakeDecision;
+    brakeDecision.BrakeAmount = 1;
+    brakeDecision.RunIndicateCommands();
+
+    auto activeIndicators = IndicatorConfig::GetInstance()->GetActiveIndicators();
+
+    ASSERT_EQ(activeIndicators[0].Action, 3);
```

Comment:
Test that activeIndicators[1] doesn't exist or that the list consists of 1 element. 
Same for the test below